x-pack/legacy/plugins/task_manager/task_runner.test.ts

```diff
@@ -655,9 +728,10 @@ describe('TaskManagerRunner', () => {
     await runner.run();
 
     if (shouldBeValid) {
-      sinon.assert.notCalled(logger.warn);
+      expect(logger.warn).not.toHaveBeenCalled();
```

Comment:
checking `logger.warn()` for calls seems very fragile, but also seems like it will be fairly obvious what's going on if it does end up breaking (because some other warning occurred in between, or whatever).  And it's been like this forever I guess :-)

Reply:
Haha yeah, I thought the same, but chose just to get rid of the sinon stuff as it was making things difficult to change.

I'll see if it's worth cleaning up as part of this.